orchestrator.py

```diff
@@ -101,16 +101,26 @@ def toggle_logging(self, _) -> None:
             topic = module.stop if self.currently_logging else module.start
             self.mqtt_client.publish(str(topic))
 
-        self.mqtt_client.publish(
-            str(
-                topics.BOOST.stop
-                if self.currently_logging
-                else topics.BOOST.start
-            )
-        )
-
-        # `self.currently_logging` will be updated when we receive the message
-        # we publish above.
```

Comment:
Is there any reason this comment wouldn't still apply?

I believe the logic is that we shouldn't call `self.set_logging_state` because
- If the `start` message was sent successfully, since we're also subscribed to the topic, `self.logging_state` will be updated in `on_message`
- If the `start` message failed to send, we don't want to update the internal logging state so that the next button press causes it to try again.

Maybe the comment in the code should have been more explicit about this reasoning - feel free to copy it in.

Reply:
I thought the reason we didn't update it there was self explanatory so I removed the comment, but I've added it back in again.

Reply:
You are updating it manually though, when you call `self.set_logging_state(next_logging_state)`

Reply:
My bad, I probably shouldn't have just copied and pasted that code. I've removed that line and I've added more verbose explanation of why we don't include it as comments.